File: tasklink_mobile/lib/services/task_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'package:shared_preferences/shared_preferences.dart';

class TaskService {
  final String baseUrl = 'http://localhost:5084/api/Tasks';

  Future<Map<String, String>> _getHeaders() async {
    SharedPreferences prefs = await SharedPreferences.getInstance();
    String? token = prefs.getString('token');
    return {
      'Content-Type': 'application/json',
      'Authorization': 'Bearer $token',
    };
  }

  Future<List<dynamic>> getAllTasks() async {
    final response = await http.get(Uri.parse(baseUrl), headers: await _getHeaders());
    if (response.statusCode == 200) {
      return jsonDecode(response.body);
    } else {
      throw Exception('Gagal mengambil data task');
    }
  }

  Future<bool> createTask(String title, String description, int projectId) async {
    final response = await http.post(
      Uri.parse(baseUrl),
      headers: await _getHeaders(),
      body: jsonEncode({
        'title': title,
        'description': description,
        'isCompleted': false,
        'projectId': projectId
      }),
    );
    return response.statusCode == 200 || response.statusCode == 201;
  }

  Future<bool> deleteTask(int id) async {
    final response = await http.delete(Uri.parse('$baseUrl/$id'), headers: await _getHeaders());
    return response.statusCode == 200;
  }
}

File: tasklink-web/src/services/taskService.js
const BASE_URL = 'http://localhost:5084/api/Tasks';

const getAuthHeaders = () => {
    const token = localStorage.getItem('token');
    return {
        'Content-Type': 'application/json',
        'Authorization': `Bearer ${token}`
    };
};

export const taskService = {
    
    getAll: async () => {
        const response = await fetch(BASE_URL, {
            method: 'GET',
            headers: getAuthHeaders()
        });
        if (!response.ok) throw new Error('Gagal mengambil data task');
        return await response.json();
    },

    
    create: async (taskData) => {
        const response = await fetch(BASE_URL, {
            method: 'POST',
            headers: getAuthHeaders(),
            body: JSON.stringify(taskData)
        });
        if (!response.ok) throw new Error('Gagal membuat task baru');
        return await response.json();
    },

    
    delete: async (id) => {
        const response = await fetch(`${BASE_URL}/${id}`, {
            method: 'DELETE',
            headers: getAuthHeaders()
        });
        if (!response.ok) throw new Error('Gagal menghapus task');
        return await response.json();
    }
};